src/main/java/application/rest/v1/configmaps/KindActionMappingProcessor.java

```diff
@@ -0,0 +1,462 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package application.rest.v1.configmaps;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import application.rest.v1.KAppNavEndpoint;
+import io.kubernetes.client.ApiClient;
+import io.kubernetes.client.ApiException;
+import io.kubernetes.client.apis.CustomObjectsApi;
+
+import com.ibm.kappnav.logging.Logger;
+
+/**
+ * KindActionMappings provide mapping rules that map a resource to a set of action configmaps.
+ * This class provides the methods that facilitates a process of getting a configmap for a given
+ * resource following the mapping rules and configmap hierarchy & precedence. 
+ */
+public class KindActionMappingProcessor {
+
+    private static final String className = KindActionMappingProcessor.class.getName();
+
+    // KindActionMapping/KAM definitions
+    private static final String KAM_PLURAL = "kindactionmappings";
+    private static final String KAM_GROUP = "actions.kappnav.io";
+    private static final String KAM_VERSION = "v1";
+    private static final String WILDCARD = "*";
+
+    private static final int MAX_PRECEDENCE = 9;
+    private static final int TOTAL_INDIVIDUAL_MAPPINGS = 4;
+
+    private static final int KSN = 3; // Kind, Subkind, name - instance specific
+    private static final int KN = 2; // Kind, name - instance specific
+    private static final int KS = 1; // Kind, Subkind - subkind specific
+    private static final int K = 0; // Kind - kind specific
+
+    private static final String METADATA_PROPERTY_NAME = "metadata";
+    private static final String NAMESPACE_PROPERTY_NAME = "namespace";
+    private static final String ITEMS_PROPERTY_NAME = "items";
+    private static final String SPEC_PROPERTY_NAME = "spec";
+    private static final String PRECEDENCE_PROPERTY_NAME = "precedence";
+    private static final String MAPPINGS_PROPERTY_NAME = "mappings";
+
+    private static final String APIVERSION_PROPERTY_NAME = "apiVersion";
+    private static final String NAME_PROPERTY_NAME = "name";
+    private static final String SUBKIND_PROPERTY_NAME = "subkind";
+    private static final String KIND_PROPERTY_NAME = "kind";
+    private static final String MAPNAME_PROPERTY_NAME = "mapname";
+
+    private String compNamespace;
+    private String compApiVersion;
+    private String compName;
+    private String compSubkind;
+    private String compKind;
+
+    public KindActionMappingProcessor(String namespace, String apiVersion, String name,
+                                      String subkind, String kind) {
+        this.compNamespace = namespace;
+        this.compApiVersion = apiVersion;
+        this.compName = name;
+        this.compSubkind = subkind;
+        this.compKind = kind;
+    }
+
+    /**
```

Comment:
Would probably be a good idea to document the hierarchy details in a block comment somewhere in here.